### ISSUE TITLE
distcc is registering against interfaces it is not listening on

### ISSUE THREAD

**Issue by GoogleCodeExporter** [distcc/distcc#64]
```
Using distcc 3.1 (compiled from source) under OpenSUSE 11.2 64 bit:

distccd is publishing zeroconf entries on interfaces it has NOT been instructed 
to listen on.

My system has 4 available interfaces:
lo, eth0, vmnet8, vmnet1 (VMWare interfaces for virtual machines).

distccd has been invoked with:
distccd --allow 10.200.0.0/16 --allow 127.0.0.1 --zeroconf --daemon --listen 
10.200.161.154 --listen 127.0.0.1 --no-detach --log-stderr --verbose

Avahi is showing:
+ vmnet8 IPv4 distcc@WIC-101119                             Distributed 
Compiler local
+ vmnet1 IPv4 distcc@WIC-101119                             Distributed 
Compiler local
+ eth0 IPv4 distcc@WIC-101119                             Distributed Compiler 
local

In other words, distcc is stating that it is listening on the 2 VMWare virtual 
interfaces. Since it has been told explicitly to listen on ONLY the eth0 and lo 
interfaces, it should NOT be registering there.

Moreover, this causes clients to fail, as they attempt to connect to the VMWare 
interfaces and are refused (since distccd isn't listening there anyway).


```

Original issue reported on code.google.com by `david.ha...@aeroflex.com` on 21 Jul 2010 at 9:16
